### PR TITLE
Removing targets should be always performed on the main thread

### PR DESCRIPTION
**Pull request by michalcichon**
Fix for this issue: https://github.com/mixpanel/mixpanel-iphone/issues/732

### CODE REVIEW

**Comment by michalcichon** [comment]
Build on CI failed. I guess this is kind of configuration problem (certificates?)

```
$ openssl aes-256-cbc -k "$ENCRYPTION_SECRET" -in scripts/profile/HelloMixpanel_Ad_Hoc.mobileprovision.enc -d -a -out scripts/profile/HelloMixpanel_Ad_Hoc.mobileprovision
bad decrypt
1817:error:06065064:digital envelope routines:EVP_DecryptFinal_ex:bad decrypt:/BuildRoot/Library/Caches/com.apple.xbs/Sources/OpenSSL098/OpenSSL098-59.60.1/src/crypto/evp/evp_enc.c:330:
The command "openssl aes-256-cbc -k "$ENCRYPTION_SECRET" -in scripts/profile/HelloMixpanel_Ad_Hoc.mobileprovision.enc -d -a -out scripts/profile/HelloMixpanel_Ad_Hoc.mobileprovision" failed and exited with 1 during .
Your build has been stopped.
```

**Comment by pchien** [comment]
Ah yeah @michalcichon I've been meaning to fix CI for github contributors but I haven't gotten around to it. I'll run the tests manually for this PR.

File: Mixpanel/MPUIControlBinding.m [hunk 1]
```diff
@@ -177,7 +177,9 @@ - (void)stop
         // remove target-action pairs
         for (UIControl *control in self.appliedTo.allObjects) {
             if (control && [control isKindOfClass:[UIControl class]]) {
-                [self stopOnView:control];
+                dispatch_async(dispatch_get_main_queue(), ^{
```
Comment (pchien) [inline Mixpanel/MPUIControlBinding.m]: Can you use `NSThread mp_safelyRunOnMainThreadSync:`? From https://github.com/mixpanel/mixpanel-iphone/blob/master/Mixpanel/NSThread+MPHelpers.m#L13. I've been trying to make that the standard way to do this in the SDK. Thanks!

Comment (michalcichon) [reply Mixpanel/MPUIControlBinding.m]: Sure, I changed it as you proposed.

**Comment by pchien** [comment]
@michalcichon thanks for doing this!

**Pull request merged**